packages/block-editor/src/components/use-matching-variation-information/index.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { getBlockType, getBlockVariations } from '@wordpress/blocks';
+
+/** @typedef {import('@wordpress/blocks').WPBlockDisplayInformation} WPBlockDisplayInformation */
+
+/**
+ * Hook used to try to find a matching block variation and return
+ * the appropriate information for display reasons. In order to
+ * to try to find a match we need to things:
+ * 1. Block's client id to extract it's current attributes.
+ * 2. A block variation should have set `isActive` prop to a proper function.
+ *
+ * If for any reason a block variaton match cannot be found,
+ * the returned information come from the Block Type.
+ *
+ * @param {string} clientId Block's client id.
+ * @return {WPBlockDisplayInformation} Block's display information.
+ *
+ */
+
+// TODO write jsdoc example
+// TODO write tests
+export default function useMatchingVariationInformation( clientId ) {
+	const { name, attributes } = useSelect(
+		( select ) => {
+			if ( ! clientId ) return {};
+			const { getBlockName, getBlockAttributes } = select(
+				'core/block-editor'
+			);
+			return {
+				name: getBlockName( clientId ),
+				attributes: getBlockAttributes( clientId ),
+			};
+		},
+		[ clientId ]
+	);
+	const blockType = getBlockType( name );
```

Comment:
Both `getBlockType` and `getBlockVariations` are only tiny wrappers over `core/blocks` datastore calls targeted mostly for plugin authors. It would be more convenient to use direct store references inside `useSelect` calls. It should allow running all the logic inside `useSelect` call.